### ISSUE TITLE
Consider adding support for type coercion

### ISSUE THREAD

**Issue by natebrunette** [tebru/gson-php#31]

**Comment by mattjanssen** [comment]
Yes, please! Is the only way to currently do this with custom type adapters? Like "StringToFloat" for example?

**Comment by natebrunette** [comment]
Currently, yes. I was struggling with how to implement it in a natural way. The two options I see are to create a new set of type adapters that do conversions is specific predetermined ways, or to pass a config value to the readers and skip throwing an exception if an unexpected type is found. This will rely on PHP casting the values to the expected value.

Edit: enabling this could conceal errors in the json as well if it can’t be done per field/class. But it gets more complicated to handle it on that level.
